File: 1436-destination-city/1436-destination-city.py
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        
        graph = defaultdict(list)

        for u, v in paths:
            graph[u].append(v)
            graph[v]
        
        print(graph)

        for city, destinations in graph.items():
            if len(destinations) == 0:
                return city
        
        

File: 1436-destination-city/1436-destination-city.java
class Solution {
    public String destCity(List<List<String>> paths) {
        
        Set<String> starts = new HashSet<>();

        for (List<String> path : paths) {
            String from = path.get(0);
            starts.add(from);
        }

        for (List<String> path : paths) {
            String to = path.get(1);
            if (!starts.contains(to)) {
                return to;
            }
        }
        return "";
    }
}